src/PhpSpec/Exception/Wrapper/InvalidCollaboratorTypeException.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Exception\Wrapper;
+
+
+class InvalidCollaboratorTypeException extends CollaboratorException
+{
+
+    public function __construct(\ReflectionParameter $parameter, \ReflectionFunctionAbstract $function)
+    {
+        $message = sprintf(
+            'Collaborator cannot be array or callable: argument %s defined in %s. ' .
```

Comment:
I would say ``Collaborator must be an object`` instead. Scalar typehints are not supported either